caddyhttp/proxy/upstream.go

```diff
@@ -287,9 +304,12 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream) error {
 }
 
 func (u *staticUpstream) healthCheck() {
+	client := http.Client{
+		Timeout: u.HealthCheck.Timeout,
+	}
```

Comment:
It'd be nice if we re-used this client for all health checks, I think the Go docs say something about reusing clients being a good thing.